dask_expr/tests/test_shuffle.py

```diff
@@ -136,3 +138,28 @@ def test_set_index(df, pdf):
 
     with pytest.raises(TypeError, match="can't be of type DataFrame"):
         df.set_index(df)
+
+
+def test_set_index_pre_sorted(pdf):
+    pdf = pdf.sort_values(by="y", ignore_index=True)
+    df = from_pandas(pdf, npartitions=10)
+    q = df.set_index("y")
+    assert_eq(q, pdf.set_index("y"))
+    result = q.simplify().expr
+    assert (
+        len(
+            set(result.find_operations(Expr))
+            - set(result.find_operations((Blockwise, FromPandas)))
+        )
+        == 0
+    )
+    q = df.set_index(df.y)
+    assert_eq(q, pdf.set_index(pdf.y))
+    result = q.simplify().expr
+    assert (
+        len(
+            set(result.find_operations(Expr))
+            - set(result.find_operations((Blockwise, FromPandas)))
+        )
+        == 0
+    )
```

Comment:
Just a nit, but you could probably just compare the sets directly `assert set(...) == set(...)`

Reply:
Or `assert all(isinstance(expr, ...) for expr in result.walk())`